### PR TITLE
Ensure source is set on GameplayClockContainer.Reset()

### PR DESCRIPTION
**Pull request by smoogipoo**
As pointed out by @bdach in https://github.com/ppy/osu/issues/12541, makes sense to me.

### CODE REVIEW

**Review by bdach — DISMISSED** [review]

File: osu.Game/Screens/Play/GameplayClockContainer.cs [hunk 1]
```diff
@@ -100,6 +100,7 @@ public virtual void Seek(double time)
         /// </summary>
         public virtual void Reset()
         {
+            ChangeSource(SourceClock);
```
Comment (frenzibyte) [inline osu.Game/Screens/Play/GameplayClockContainer.cs:103]: better with the comment from the other usage:
```suggestion
            // Ensure that the source clock is set.
            ChangeSource(SourceClock);

```

Comment (bdach) [reply osu.Game/Screens/Play/GameplayClockContainer.cs:103]: would rather factor out a one-liner method at that point if we're nitpicking to this degree.

Comment (peppy) [reply osu.Game/Screens/Play/GameplayClockContainer.cs:103]: also make the comment mention why it possibly wouldn't be already set. looking at this PR in isolation makes me raise an eyebrow, so a comment may be good to have, explaining what this requirement comes from

Comment (bdach) [reply osu.Game/Screens/Play/GameplayClockContainer.cs]: Have made a change that attempts to explain better. Please advise if it's any better.

**Comment by bdach** [comment]
While running tests locally I noticed that `TestSceneMasterGameplayClockContainer.TestElapseThenReset` was failing here, but apparently didn't fail CI, and doesn't fail on `master`. So I'm a bit suspicious of this diff now, even though I proposed this change myself. Needs more investigation, I think...

**Review by peppy — DISMISSED** [review]

**Comment by bdach** [comment]
I believe I have fixed the failures. The last commit message explains what was going on, but in short:

* while decoupleable clock is running, its current time doesn't come from either the source or the stopwatch - it comes from the base interpolating clock
* a source change of a decoupleable seeks the source clock also
* when an interpolating clock is seeked, its interpolation state (last/current interpolated time) are reset to 0
* that triggers [this branch](https://github.com/ppy/osu-framework/blob/master/osu.Framework/Timing/InterpolatingFramedClock.cs#L91-L95) in the interpolating clock, causing the current time of the interpolating clock to get stuck at the time at the point of the source change, until the framed source clock catches up.

I don't believe this would ever show up in a non-test usage (since in game `Reset()` is only ever used in the gameplay start process), but it *was* an actual issue if the reset was preceded by a `Start()` call. Not entirely sure why the failure didn't pop up on CI, but I've spent enough time on figuring this stuff out anyway for me to want to tap out for now.

**Review by peppy — APPROVED** [review]

**Pull request merged**